mod.ts

```diff
@@ -131,6 +131,16 @@ export interface BuildOptions {
      * @default true
      */
     skipLibCheck?: boolean;
+    /**
+     * Deno enabled `isolatedModules` by default. but you can config `isolatedModules` for your js output
+     * @default true
+     */
+    isolatedModules?: boolean;
```

Comment:
> isolatedModules chould [references to const enum members](https://www.typescriptlang.org/tsconfig#references-to-const-enum-members), is an looser options.
> we should open this options, should work~

Do you really need this compiler option? I'd rather not enable or allow this one because supporting const enums might introduce some unforseen complexity in the future.

Reply:
> > isolatedModules chould [references to const enum members](https://www.typescriptlang.org/tsconfig#references-to-const-enum-members), is an looser options.
> > we should open this options, should work~
> 
> Do you really need this compiler option? I'd rather not enable or allow this one because supporting const enums might introduce some unforseen complexity in the future.

you are right.
I open this options. just because maybe if some one want to use tc39 decorators in the future, the compileOptions maybe: `target: "esnext", experimentalDecorators: false`.

Reply:
> Do you really need this compiler option? I'd rather not enable or allow this one because supporting const enums might introduce some unforseen complexity in the future.

For developers, since we are converting the deno code into a standard typescript project, and then using ts-morph to compile the code into esm with cjs to be compatible with npm.
This process is irreversible, so I just want to expose more compilation options to make it more compatible with the developer's needs.

For me, the purpose of writing `const enum` instead of `enum` is zero cost abstraction. That's why I'm opening up this option.

If you think it has unpredictable risks, I think that should also be a decision for the user to make. Rather than letting the tool restrict it, after all the default value of true is enough for most novice users to avoid the problems you are worried about.